frontend/lib/resolution/default-functions.cpp

```diff
@@ -293,40 +294,209 @@ static void buildInitArgs(Context* context,
   }
 }
 
+static void collectFields(const AstNode* ast,
+                          std::vector<const VarLikeDecl*>& fields) {
+  if (auto decl = ast->toAggregateDecl()) {
```

Comment:
I suspect that this won't do what you want with a record containing a nested record. (Would that pattern get this far?)

Reply:
Good catch! Fixed.